crates/task-impls/src/quorum_vote/handlers.rs

```diff
@@ -83,6 +81,28 @@ pub(crate) async fn handle_quorum_proposal_validated<
         .await
     };
 
+    if version >= V::Epochs::VERSION {
+        if let Some(last_leaf_view) = leaf_views.last() {
+            let decided_block_number = last_leaf_view.leaf.block_header().block_number();
+
+            let current_epoch_number = TYPES::Epoch::new(epoch_from_block_number(
+                decided_block_number,
+                task_state.epoch_height,
+            ));
```

Comment:
We can get `current_epoch_number` from `proposal.block_header.block_number()`, so there's no need to use the leaf or decided block info. 

This can also be moved to before the `let LeafChainTraversalOutcome`.